pkg/installmanager/actuator.go

```diff
@@ -0,0 +1,16 @@
+package installmanager
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Actuator interface is the interface that is used to add cloud provider support to the deprovision controller.
+type Actuator interface {
```

Comment:
WDYT about LogUploader here? I don't love the idea of using the actuator terminology because we're doing something so different here. We use actuators where we fork behavior for a cluster depending on the cloud it's on, and I think typically when we're doing a desired/current reconciliation.

In this case we're checking if a log upload storage is configured for multiple implementations and telling it to upload something, we might use s3 upload for gcp/azure/any cloud provider clusters provisioned in a Hive that's running on AWS, or even on any platform to minio.